src/ClipMirAudioReader.cpp

```diff
@@ -0,0 +1,40 @@
+#include "ClipMirAudioReader.h"
+#include "ClipInterface.h"
+
+#include <cassert>
+
+ClipMirAudioReader::ClipMirAudioReader(const ClipInterface& clip)
+    : mClip(clip)
+{
+}
+
+int ClipMirAudioReader::GetSampleRate() const
+{
+   return mClip.GetRate();
+}
+
+long long ClipMirAudioReader::GetNumSamples() const
+{
+   return mClip.GetVisibleSampleCount().as_long_long();
+}
+
+void ClipMirAudioReader::ReadFloats(
+   float* buffer, long long where, size_t numFrames) const
+{
+   AddChannel(0, buffer, where, numFrames);
```

Comment:
Previous contents of `buffer` matter -- this is addition, not assignment.  A header comment should say that.

Reply:
I took another look this morning, and I'm still not convinced, that the loop using `GetNextFrame` in `MirDsp.cpp` is correct.  I still think later passes have a buffer contaminated with previous data.

But all of that could be fixed down at this level, if only the first `AddChannel` instead assigned the buffer, not accumulating into it.  Only the second `AddChannel` should remain.  And then no special header comments would be needed.

Reply:
Yes indeed, this wasn't intended behaviour. Addressing.

Reply:
I see it is now corrected, though there are two passes over the buffer (to zero it, then to add to it).  Maybe that marginal inefficiency is unimportant.